### PR TITLE
Balance Tracking Fixes and Construction Endpoints

### PR DESCRIPTION
**Pull request by nddeluca**
- Fixes balance tracking and passes check:data on mainnet
- Adds construction metadata, parse, payloads, and combine endpoints
- Adds configuration for rosetta-cli check:construction
- Bumps default gas adjustment to 150% of gas estimated.  Tried 110% and 120%, but was not reliable enough.

### CODE REVIEW

File: services/errors.go [hunk 1]
```diff
@@ -28,6 +28,15 @@ var (
...
+		ErrInvalidCurrencyAmount,
+		ErrUnsupportedCurrency,
+		ErrUnclearIntent,
+		ErrInvalidAddress,
+		ErrInvalidOptions,
+		ErrInvalidMetadata,
+
 		ErrUnsupportedCurveType,
 		ErrPublicKeyNil,
 		ErrInvalidPublicKey,
```
Comment (karzak) [inline services/errors.go:42]: `ErrMissingPublicKey` is not included in this list, but is defined on L131

**Review by karzak — APPROVED** [review]
Few nits, otherwise LGTM

File: kava/client.go [hunk 2]
```diff
@@ -307,20 +338,40 @@ func (c *Client) getOperationsForTransaction(
...
 	result *abci.ResponseDeliverTx,
 ) []*types.Operation {
-	var status string
+	var opStatus string
 
 	if result.Code == abci.CodeTypeOK {
-		status = SuccessStatus
+		opStatus = SuccessStatus
+	} else {
+		opStatus = SuccessStatus
```
[the remark this answers is not in the document]
Comment (nddeluca) [reply kava/client.go]: thx good catch, refactored out the unnecessary branches w/ initialization

File: services/errors.go (hunk 1 above)
Comment (nddeluca) [reply services/errors.go:43]: 👍

**Pull request merged**